src/auth/AppAuth0ClientPromise.tsx

```diff
@@ -31,7 +31,7 @@ const finalAuthConfig = {
  */
 const AppAuth0ClientPromise: Promise<Auth0Client> =
   // Note: The window variable is not defined static rendering.
-  typeof window === "undefined"
+  typeof window.crypto === "undefined"
```

Comment:
I've updated this if statement (not sure if ifs are statements or expressions in Javascript/Typescript) to match for `window.crypto` instead of `window` because `window` is emulated by Jest (using JSDom) when tests are run but Jest doesn't include code for `window.crypto` which Auth0 relies upon.

Reply:
Before going forward with this change, since it's auth-related and high stakes, I'd like @szhu to weigh in and see if he has any concerns.

Reply:
Cool! (I think `window.crypto === "undefined"` should be true if `window === "undefined"`)

Reply:
I think this is fine as long as you document in a code comment:

1.  Why you are testing for `window.crypto`
2.  How the user can manually test that things that this trick depends on still hold true in the future

For example, the existing comment, "The window variable is not defined static rendering." is a statement that can easily be tested in the future. (However, I am missing a statement about why this check is there, which should be something like "The Auth0 client code breaks during static render, but we also don't need it during static render." It would be nice if you can help me add that in as well.)

---

Here are 3 specific things that you can follow to meet the goals above, but if you have another way to address these concerns feel free! Also happy to discuss options here and on Slack.

1.  Why do we need to skip the Auth0 stuff during testing?
2.  How did you check (and how can someone in the future check) that `window.crypto` is undefined during testing?
3.  Please integrate these comments with the comment about static render, such that it makes sense as a whole.

Here's my best guess at what a good final comment will look like, feel free to use this and correct where I'm wrong:

> We need to not use the Auth0 client...
> 
> *   during testing, because it breaks under this environment and because our current test code is not testing auth-related functionality, or
> *   during static render, because it breaks under this environment and because the auth state is different for each visitor anyway.
> 
> We use `typeof window.crypto === "undefined"` because this works to detect both the testing and static render environments. (\`typeof window === undefined\` only detects the static render environment.